src/graph.rs

```diff
@@ -39,6 +43,62 @@ impl<'a, N: Clone, E: Clone, Ty: EdgeType> Graph<N, E, Ty> {
         })
     }
 
+    pub fn edge_by_screen_pos(
+        &self,
+        meta: &'a Metadata,
+        style: &'a SettingsStyle,
+        screen_pos: Pos2,
+    ) -> Option<(EdgeIndex, &Edge<E>)> {
+        let pos_in_graph = (screen_pos.to_vec2() - meta.pan) / meta.zoom;
+        self.edges_iter().find(|(idx, e)| {
+            let (source, target) = self.g.edge_endpoints(*idx).unwrap();
+            let source_pos = self.g.index(source).location();
+            let target_pos = self.g.index(target).location();
+            if source == target { // edge is a loop (bezier curve)
```

Comment:
Order should be used here to select proper looped edge from the set of duplicates